src/tools/tests/allocator-recipe-resolver-test.ts

```diff
@@ -1562,4 +1562,80 @@ recipe ThingWriter
     assert.equal(handle.getTtl().toDebugString(), '3d');
   });
 
+  // In the tests below, `AnotherThing` will have restrictions in the policy and ingress
+  // validation should allow it to be read.
+  const outsideOfPolicyReads = `
+schema Thing {a: Text, b: Text, c: Text, d: Text}
+schema AnotherThing {w: Text, x: Text, y: Text, z: Text}
+
+particle ThingWriter
+  thingOut: writes [Thing]
+particle ThingReader
+  thingIn: reads [Thing {a, b, c}]
+  anotherOut: writes [AnotherThing {w, x, y, z}]
+particle AnotherReader
+  anotherIn: reads [AnotherThing]
+
+@arcId('test')
+recipe R
+  thing: create 'my-handle-id' @persistent @ttl('2d')
+  another: create 'my-out-id' @persistent @ttl('2d')
+  ThingWriter
+    thingOut: writes thing
+  ThingReader
+    thingIn: reads thing
+    anotherOut: writes another
+  AnotherReader
+    anotherIn: reads another
+`;
+
+  it('does not allow reads if schema is not mentioned in policy', async () => {
```

Comment:
nit - each schema Thing and AnotherThing have 4 fields, and i think for the given tests 2 fields would have been enough. I think having less fields is possible in general makes the test more readable.

Reply:
done.